### ISSUE TITLE
Removal of BinaryStore parameter has broken our helm installs

### ISSUE THREAD

**Issue by jmak0010** [jfrog/charts#1723]
https://github.com/jfrog/charts/blob/6a681ad06c651f9f377dcd3fe8628cde6f560c45/stable/artifactory-ha/values.yaml#L839

Hello Jfrog,

We utilize this value in our helm charts for our production artifactory-ha installation to pass a custom BinaryStore.xml to the pods. We also use the above commands to Copy File on Startup every time.

Since you have removed this value, the primary pod startup now fails with the following error on the latest helm chart version (107.49.5):

> cp: cannot stat '/artifactory_bootstrap/binarystore.xml': No such file or directory

Please advise on how we can pass a custom binarystore.xml copy to the helm chart in current versions. In the referenced files/binarystore I only see options for your precanned choices of binary store.

**Comment by jmak0010** [comment]
Also, pinning the old chart version that we used prior to this change (107.41.12) now fails to start up with a failure to read platform configuration. The Primary pod never starts, and the member pods never even initialize. Are there configuration changes to the platform config files from the new helm charts that are not backwards compatable?

**Comment by jmak0010** [comment]
It would be great if this file could be enhanced to allow for another option of passing in a fully custom binarystore.xml file through the old parameter.
https://github.com/jfrog/charts/blob/6a681ad06c651f9f377dcd3fe8628cde6f560c45/stable/artifactory-ha/files/binarystore.xml#L1

You could use If logic to test if the binarystoreXml parameter is set, and if it is take it's value instead of the if/else logic that follows with the hard-coded options of persistence.

**Comment by jmak0010** [comment]
you also left the actual secret file completely alone which is alarming as without the values.yaml entry for binarystore this will ALWAYS result in a null variable on this line.

https://github.com/jfrog/charts/blob/6a681ad06c651f9f377dcd3fe8628cde6f560c45/stable/artifactory/templates/artifactory-binarystore-secret.yaml#L13

Can you provide any reasoning why this variable was removed?

**Comment by srinivasgowda097** [comment]
@jmak0010  Thanks for raising this issue

Here is the quick history of binarystoreXml configuration:

- Prior to Artifactory chart versions 107.46.x, Entire binarystoreXml configuration was in values.yaml( which was spanned over 300 lines), we wanted to refactor this part of configuration and move it under `files/binarystore.xml`.

- For Artifactory chart versions >= 107.46.x, If there is custom configuration of binarystore.xml in values.yaml (`.Values.artifactory.persistence.binarystoreXml`), users can still use this configuration, so nothing changes for user if they maintain custom configuration in values.yaml.

-  From Artifactory chart versions >=7.46.x, `copyOnEveryStartup` is not valid anymore for binarystoreXml config, it is always copied via initContainers, so you can remove `copyOnEveryStartup` in values.yaml

Hope this answers your queries, Please reach out to us if you have further questions

**Comment by jmak0010** [comment]
> From Artifactory chart versions >=7.46.x, copyOnEveryStartup is not valid anymore for binarystoreXml config, it is always copied via initContainers, so you can remove copyOnEveryStartup in values.yaml

This comment seems to be the answer, after removing the copyOnEveryStartup section the pod no longer errors on the file copy step. I then confirmed that the input binarystoreXml is the correct custom file from my helm values.

However now my pods fail to migrate their configuration to the new version due to a licensing issue. I have added details on this new error to my open Jfrog support case.

With this additional information we can close this github issue regarding the BinaryStore parameter. Thank you for the response and information, especially about removing the copyOnEveryStartup. I might suggest adding a comment to the values.yaml file noting that it is always copied now, and that those values are not needed to be set, as they will cause the error I listed above.

> cp: cannot stat '/artifactory_bootstrap/binarystore.xml': No such file or directory
